akka-actor/src/main/scala/akka/actor/ActorSystem.scala

```diff
@@ -897,15 +897,27 @@ private[akka] class ActorSystemImpl(
 
   def actorOf(props: Props, name: String): ActorRef =
     if (guardianProps.isEmpty) guardian.underlying.attachChild(props, name, systemService = false)
-    else
-      throw new UnsupportedOperationException(
-        s"cannot create top-level actor [$name] from the outside on ActorSystem with custom user guardian")
+    else {
+      val message =
+        if (isTypedGuardian)
+          s"cannot create top-level actor [$name] from the outside on a typed ActorSystem.  A classic ActorSystem " +
+          "should be created instead and used to spawn typed actors."
```

Comment:
```suggestion
          s"cannot create top-level actor [$name] from the outside on a typed ActorSystem. "A classic ActorSystem can be used instead, allowing both typed actors and spawning top level actors with 'actorOf'."
```

SpawnProtocol is also an option if you are in control of the code that wants to start actors from "the outside" but maybe the error message is long enough as it is.

Reply:
`SpawnProtocol` doesn't really support classic actors, AFAIK, which is what was being attempted here (though admittedly my exception message doesn't completely capture that).

Reply:
I think "created" is better verbiage than "used" because type-wise `classicSystem` _is_ a classic ActorSystem (we're arguably in LSP hot water...).  I don't think anyone is going to create a classic ActorSystem with a custom guardian in response to this message (that's a kind of obscure trick, AFAICT), so creating a classic ActorSystem vs. a typed ActorSystem is what to do if you need a top-level actor.

```
cannot create top-level actor from the outside on a typed ActorSystem.  A classic ActorSystem should be created in order to spawn top-level actors.
```

Reply:
So what about:
"A classic ActorSystem can be created instead, to allow both typed actors and spawning top level actors with 'actorOf'."

The reason for avoiding "should" is that it can also make just as much sense to avoid trying to spawn top level actors and instead follow the typed-idiomatic guardian spawning children pattern in a given application.

Reply:
I think it's more likely than not that this is the result of pulling in a classic extension or third-party library which hasn't been updated for typed, in which case the classic ActorSystem is a "should" (similar would be searching for "akka udp", which for me shows the Classic Networking docs above Alpakka UDP (see #27263))

Reply:
I think it's just the right side of "too long"

Reply:
Ah, missed the last push. I like this long error best, it will anyway happen once and the system will be broken, or it won't happen at all.